lib/improver/cli/nbhood.py

```diff
@@ -347,13 +140,12 @@ def process(cube, neighbourhood_output, neighbourhood_shape, radius=None,
             neighbourhood_output='percentiles'.
         ValueError:
             If degree_as_complex is used with neighbourhood_shape='circular'.
-        ValueError:
-            If degree_as_complex is used with apply_recursive_filter.
-        ValueError:
-            If neighbourhood_shape is 'circular' and apply_recursive_function
-            is True.
-
     """
+    from improver.nbhood.nbhood import (
+        GeneratePercentilesFromANeighbourhood, NeighbourhoodProcessing)
+    from improver.utilities.pad_spatial import remove_cube_halo
+    from improver.wind_calculations.wind_direction import WindDirection
+
     if (neighbourhood_output == "percentiles" and
             neighbourhood_shape == "square"):
         raise RuntimeError('neighbourhood_shape="square" cannot be used with'
```

Comment:
The simplest way to set this up from a user perspective is to throw an error if `neighbourhood_shape == "square"` and `neighbourhood_output` is set.  This means all the calls to square neighbourhooding don't have to set this extra argument, as they always only output probabilities.  But this is something that might want to change in all the neighbourhooding CLIs, so I could look at that in refactoring later if it's messy to do at this stage.

Reply:
I reordered the tests to group them slightly better. Removed an extraneous test, and I think I will leave it there.